### PR TITLE
restructured API response

### PR DESCRIPTION
**Pull request by Amamgbu**

### CODE REVIEW

File: model/node_differ.py [hunk 1]
```diff
@@ -134,51 +134,47 @@ def get_diff_count(result):
...
                 curr_text = result["sections-curr"][c["curr"]["section"]]
-                curr_text = curr_text[c["curr"]['offset']:c["curr"]['offset'] + c["curr"]['size']].replace("\n", "\\n")
-                is_edit_type_found, wikitext, edit_type = is_edit_type(prev_text, c['prev']['type'])
-                # check if edit_type in edit types dictionary
-                if edit_type in edit_types.get('change').get('edit_types').keys() and is_edit_type:
-                    edit_types['change']['edit_types'][edit_type] += 1
+                curr_text = curr_text[c["curr"]['offset']:c["curr"]['offset']+c["curr"]['size']].replace("\n", "\\n")
+                is_edit_type_found,wikitext,edit_type = is_edit_type(prev_text,c['prev']['type'])
+                #check if edit_type in edit types dictionary
+                if edit_types.get(edit_type,{}) and is_edit_type_found:
```
Comment (geohci) [inline model/node_differ.py]: I'm not sure the `is_edit_type_found` variable is having an effect right now. My understanding is that if it's False, you probably don't want to do increment any counts (just continue). If I'm reading the code right, currently if it's False, it still leads to a clause that updates the edit_types dictionary. So maybe instead just add something like: `if is_edit_type_found: continue` after running the `is_edit_type` function. Then you don't have to include it in any of the other if/else statements. This applies to each of the insert/remove/change clauses (not just this specific one).

File: model/node_differ.py [hunk 2]
```diff
@@ -134,51 +134,47 @@ def get_diff_count(result):
...
+                    if edit_types.get(edit_type,{}).get('change'):
```
Comment (geohci) [inline model/node_differ.py]: this whole if/else block can be simplified I think to the following: `edit_types[edit_type]['change'] = edit_types[edit_type].get('change', 0) + 1`
Again, same goes for the corresponding insert/remove clauses higher up in the code

**Review by geohci — COMMENTED** [review]
Just two small comments that I think should be easy to address. Once they're complete, you can then focus about adapting to the new API structure! Thanks!

File: model/node_differ.py (hunk 2 above)
Comment (Amamgbu) [reply model/node_differ.py]: Thanks for the review, Isaac! I'd implement these changes.

**Comment by geohci** [comment]
Some quick high-level thoughts:
* Thanks for making the requested changes from the first PR set of comments
* I'm okay with the `.gitignore` file being added if that's helpful (though in some ways, that might be better left as local because it's pretty environment-specific). This push includes changing the license and some `.cfg` and `.pyc` files though that I don't think belong. I assume the license was a mistake, but if there's a reason to change the license, let me know. Otherwise, please remove those from the PR.
* I have some suggestions for the `is_edit_type` and `is_change_in_edit_type` but I think it might be best to merge in these changes as is (once the extraneous files are removed) and then address these two functions in a separate PR so that this one doesn't grow too large / block some of these smaller changes from being pushed through.

**Comment by Amamgbu** [comment]
> Some quick high-level thoughts:
> 
> * Thanks for making the requested changes from the first PR set of comments
> * I'm okay with the `.gitignore` file being added if that's helpful (though in some ways, that might be better left as local because it's pretty environment-specific). This push includes changing the license and some `.cfg` and `.pyc` files though that I don't think belong. I assume the license was a mistake, but if there's a reason to change the license, let me know. Otherwise, please remove those from the PR.
> * I have some suggestions for the `is_edit_type` and `is_change_in_edit_type` but I think it might be best to merge in these changes as is (once the extraneous files are removed) and then address these two functions in a separate PR so that this one doesn't grow too large / block some of these smaller changes from being pushed through.

That was a mistake. I'd fix and make a commit.

**Comment by geohci** [comment]
Thanks! Merging

**Pull request merged**